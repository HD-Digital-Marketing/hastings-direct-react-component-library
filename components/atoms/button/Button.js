import React from 'react'
import classNames from 'classnames'
import PropTypes from 'prop-types'
import './button.scss'

const Button = props => {

    const {
        type = 'button',
        url = '',
        className = '',
        theme = '',
        children,
        onClick
    } = props

    const buttonClass = classNames('a-btn', className, {
        'a-btn--primary': theme === 'primary',
        'a-btn--secondary': theme === 'secondary'
    })

    const isAnchor = url && (url.includes('http') || url.startsWith('#') || url.startsWith('mailto') || url.startsWith('/'))

    const renderAnchor = () => {
        <a href={url} className={buttonClass}>{children}</a>
    }

    const renderButton = () => {
        <button {...{ type, onClick }} className={buttonClass}>{children}</button>
    }

    return (
        isAnchor ? renderAnchor() : renderButton()
    )
}

Button.propTypes = {
  type: PropTypes.string,
  url: PropTypes.string,
  className: PropTypes.string,
  theme: PropTypes.string,
  children: PropTypes.string,
  onClick: PropTypes.string
}

export default Button
