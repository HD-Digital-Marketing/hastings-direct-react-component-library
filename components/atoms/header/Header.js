import React from 'react'
import './header.scss'
import classNames from 'classnames'
import PropTypes from 'prop-types'

const Header = props => {

    const {
        children,
        headerSize = 'h1',
        isReg = false,
        isPostcode = false,
        isSpecial = false,
        className = '',
        theme
    } = props

    let header = ''

    let headerClass = classNames('a-header', className, {
        'a-header--primary': theme === 'primary',
        'a-header--primary-xl': theme === 'primary-xl',
        'a-header--primary-on-dark': theme === 'primary-dark',
        'a-header--primary-on-dark-xl': theme === 'primary-dark-xl',
        'a-header--secondary': theme === 'secondary',
        'a-header--secondary-xl': theme === 'secondary-xl',
        'a-header--secondary-on-dark': theme === 'secondary-dark',
        'a-header--secondary-on-dark-xl': theme === 'secondary-dark-xl',
        'a-header--reg': isReg,
        'a-header--postcode': isPostcode,
        'a-header--special': isSpecial
    })

    switch (headerSize) {
        case 'h1':
            header = <h1 className={headerClass}>{children}</h1>
            break;
        case 'h2':
            header = <h2 className={headerClass}>{children}</h2>
            break;
        case 'h3':
            header = <h3 className={headerClass}>{children}</h3>
            break;
        case 'h4':
            header = <h4 className={headerClass}>{children}</h4>
            break;
        case 'h5':
            header = <h5 className={headerClass}>{children}</h5>
            break;
        case 'h6':
            header = <h6 className={headerClass}>{children}</h6>
            break;
        default:
            header = <h1 className={headerClass}>{children}</h1>
    }

    return (
        header
    )

}

Header.propTypes = {
    children: PropTypes.node,
    headerSize: PropTypes.string,
    isReg: PropTypes.bool,
    isPostcode: PropTypes.bool,
    isSpecial: PropTypes.bool,
    className: PropTypes.string,
    theme: PropTypes.string
}

export default Header