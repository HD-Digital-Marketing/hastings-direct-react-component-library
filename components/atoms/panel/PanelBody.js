import React from 'react'
import classNames from 'classnames'
import PropTypes from 'prop-types'

const PanelBody = props => {

    const {
        centerContent = false,
        className = '',
        children
    } = props

    const panelBodyClass = classNames('m-panel__body', className, className, {
        'm-panel__body--centered-content': centerContent
    })

    return(
        <div className={panelBodyClass}>{children}</div>
    )
}

PanelBody.propTypes = {
    className: PropTypes.string,
    children: PropTypes.node
}

export default PanelBody
