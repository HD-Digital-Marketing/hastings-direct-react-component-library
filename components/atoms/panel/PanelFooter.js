import React from 'react'
import classNames from 'classnames'
import PropTypes from 'prop-types'

const PanelFooter = props => {

    const {
        centerContent = false,
        className = '',
        children
    } = props

    const panelFooterClass = classNames('m-panel__footer', className, className, {
        'm-panel__footer--centered-content': centerContent
    })

    return(
        <div className={panelFooterClass}>{children}</div>
    )
}

PanelFooter.propTypes = {
    className: PropTypes.string,
    children: PropTypes.node
}

export default PanelFooter
