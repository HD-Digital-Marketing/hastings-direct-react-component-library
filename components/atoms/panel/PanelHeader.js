import React from 'react'
import classNames from 'classnames'
import PropTypes from 'prop-types'

const PanelHeader = props => {

    const {
        centerContent = false,
        className = '',
        children
    } = props

    const panelHeaderClass = classNames('m-panel__header', className, {
        'm-panel__header--centered-content': centerContent
    })

    return(
        <div className={panelHeaderClass}>{children}</div>
    )
}

PanelHeader.propTypes = {
    className: PropTypes.string,
    children: PropTypes.node
}

export default PanelHeader
