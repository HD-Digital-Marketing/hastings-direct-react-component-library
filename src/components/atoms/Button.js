import React from 'react';

const Button = (props) => {

    const {
        children,
        type = 'button'
    } = props;

    return(
        <button type={type}>{children}</button>
    )

}

export default Button;