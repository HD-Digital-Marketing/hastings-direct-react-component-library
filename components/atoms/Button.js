import React from 'react';
import '../../scss/atoms/button.scss'

const Button = (props) => {

    const {
        children,
        type = 'button'
    } = props;

    return(
        <button className="a-btn" type={type}>{children}</button>
    )

}

export default Button;