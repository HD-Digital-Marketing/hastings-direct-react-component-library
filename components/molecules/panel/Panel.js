import React from 'react'
import classNames from 'classnames'
import Button from '../../atoms/button/Button'
import Link from '../../atoms/link/Link'

const Panel = props => {

    const {
        children,
        className = '',
        hasButton = false,
        hasLink = false
    } = props

	const panelClass = classNames('m-panel', className)

    const renderButton = () =>
        <div className='m-panel__btn'>
            <Button theme='primary'>Button</Button>
        </div>

    const renderLink = () =>
        <div className='m-panel__link'>
            <Link href='https://www.hastingsdirect.com/'>Link Example</Link>
        </div>

	return (
		<div className={panelClass}>
            <p>{children}</p>
			{hasButton ? renderButton() : null}
            {hasLink ? renderLink() : null}
		</div>
	)
}

export default Panel