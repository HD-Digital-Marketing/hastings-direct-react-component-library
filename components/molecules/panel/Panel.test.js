import React from 'react'
import renderer from 'react-test-renderer'
import Panel from './Panel'
import PanelHeader from '../../atoms/panel/PanelHeader'
import Header from '../../atoms/header/Header'
import PanelBody from '../../atoms/panel/PanelBody'
import PanelFooter from '../../atoms/panel/PanelFooter'
import Button from '../../atoms/button/Button'

describe('Check the panel renders', () => {
    test('it should match the snapshot', () => {

        const tree = renderer
            .create(
                <Panel theme="dark" centerContent={true}>
                    <PanelHeader>
                        <Header headerType="h2">This is a panel title with centered content</Header>
                    </PanelHeader>
                    <PanelBody>
                        <p>Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua. Ut enim ad minim veniam, quis nostrud exercitation ullamco laboris nisi ut aliquip ex ea commodo consequat.</p>
                        <p>Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua. Ut enim ad minim veniam, quis nostrud exercitation ullamco laboris nisi ut aliquip ex ea commodo consequat.</p>
                    </PanelBody>
                    <PanelFooter>
                        <Button theme="primary">A panel button</Button>
                    </PanelFooter>
                </Panel>
            )
            .toJSON()
        expect(tree).toMatchSnapshot()

    })
})