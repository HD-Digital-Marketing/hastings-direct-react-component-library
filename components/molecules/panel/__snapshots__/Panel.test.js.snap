// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Check the panel renders it should match the snapshot 1`] = `
<div
  className="m-panel m-panel--dark m-panel--centered-content"
>
  <div
    className="m-panel__header"
  >
    <h1
      className="a-header"
    >
      This is a panel title with centered content
    </h1>
  </div>
  <div
    className="m-panel__body"
  >
    <p>
      Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua. Ut enim ad minim veniam, quis nostrud exercitation ullamco laboris nisi ut aliquip ex ea commodo consequat.
    </p>
    <p>
      Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua. Ut enim ad minim veniam, quis nostrud exercitation ullamco laboris nisi ut aliquip ex ea commodo consequat.
    </p>
  </div>
  <div
    className="m-panel__footer"
  >
    <button
      className="a-btn a-btn--primary"
      type="button"
    >
      A panel button
    </button>
  </div>
</div>
`;
